common/models/user.js

```diff
@@ -564,7 +564,11 @@ module.exports = function(User) {
     UserModel.on('attached', function() {
       UserModel.afterRemote('confirm', function(ctx, inst, next) {
         if (ctx.req) {
-          ctx.res.redirect(ctx.req.param('redirect'));
+          // Find redirect parameter as param or query. Not in body.
+          var redirectTo = (ctx.req.params && ctx.req.params.redirect) ||
+            (ctx.req.query && ctx.req.query.redirect);
```

Comment:
Why the redirect parameter cannot be in the body? Unless you can prove that such request would not be valid, or that such usage is explicitly forbidden by the docs, then this patch must check `req.body` too in the same way as `req.param()` does.